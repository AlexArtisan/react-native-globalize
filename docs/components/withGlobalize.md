# `withGlobalize`

Higher-order component (HOC) to inject the Globalize object as a prop.

This component is mainly included to maintain backwards compatibility as the `useGlobalize` hook is now the recommended way to access React Native Globalize's formatting functionality. This component itself uses `useGlobalize` to get access to the Globalize object for injecting. However, while it is not favored compared to the hook, `withGlobalize` is NOT currently deprecated.

## Usage

```js
import React from 'react';
import { withGlobalize } from 'react-native-globalize';

const ExampleComponent = ({ globalize }) => {
  // Prop `globalize` is the same as returned by `useGlobalize`
  const { formatCurrency } = globalize;
};

export default withGlobalize(ExampleComponent);
```
