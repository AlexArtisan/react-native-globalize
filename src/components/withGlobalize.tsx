/*!
 * React Native Globalize
 *
 * Copyright 2015-2020 Josh Swan
 * Released under the MIT license
 * https://github.com/joshswan/react-native-globalize/blob/master/LICENSE
 */

import React, { useContext } from 'react';
import { Globalize } from '../globalize';
import { GlobalizeContext } from '../context';

/* eslint-disable react/jsx-props-no-spreading */

export interface WithGlobalizeProps {
  globalize: Globalize;
}

export const withGlobalize = <P extends WithGlobalizeProps>(
  Component: React.ComponentType<P>,
): React.FC<Omit<P, keyof WithGlobalizeProps>> => (props) => {
    const globalize = useContext(GlobalizeContext);

    return <Component {...props as P} globalize={globalize} />;
  };
