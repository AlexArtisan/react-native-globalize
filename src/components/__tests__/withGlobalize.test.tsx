/*!
 * React Native Globalize
 *
 * Copyright 2015-2020 Josh Swan
 * Released under the MIT license
 * https://github.com/joshswan/react-native-globalize/blob/master/LICENSE
 */

import React from 'react';
import { Text } from 'react-native';
import { createWithGlobalize } from '../../../test/createWithGlobalize';
import { withGlobalize, WithGlobalizeProps } from '../withGlobalize';

describe('withGlobalize()', () => {
  test('injects Globalize instance', () => {
    const Component: React.FC<WithGlobalizeProps> = ({ globalize }) => {
      expect(typeof globalize).toBe('object');
      expect(globalize).toHaveProperty('getCurrencyFormatter');
      expect(globalize).toHaveProperty('formatCurrency');

      const formatter = globalize.getCurrencyFormatter();

      return (
        <Text>
          {formatter(10)}
        </Text>
      );
    };

    const WrappedComponent = withGlobalize(Component);

    const tree = createWithGlobalize(<WrappedComponent />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
