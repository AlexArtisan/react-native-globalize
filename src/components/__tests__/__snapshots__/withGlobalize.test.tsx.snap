// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withGlobalize() injects Globalize instance 1`] = `
<Text>
  $10.00
</Text>
`;
